generator/src/main/java/com/scottlogic/deg/generator/decisiontree/ConstraintNode.java

```diff
@@ -198,15 +114,9 @@ static ConstraintNode merge(Iterator<ConstraintNode> constraintNodeIterator) {
 
             atomicConstraints.addAll(constraintNode.getAtomicConstraints());
             decisions.addAll(constraintNode.getDecisions());
-            markings.addAll(constraintNode.getNodeMarkings());
+            markings.addAll(constraintNode.nodeMarkings);
```

Comment:
In general I prefer to avoid directly accessing fields, and like having getters. See if you agree with https://stackoverflow.com/a/1568230/7470360

Reply:
this is its own private class.

Reply:
Even so I can think of a few cases where it would be useful to have private getters debugged ect.